### ISSUE TITLE
Error when installing pycisTopic in Python 3.10.8 & 3.11.0

### ISSUE THREAD

**Issue by ShaoZhiting24** [aertslab/pycisTopic#49]
Hi, I met this error when I was installing pycisTopic in Python in Python.
```
git clone https://github.com/aertslab/pycisTopic.git
cd pycisTopic
pip install .
```
Error: CRITICAL: Python version must ≥ 3.6!

My Python version used in trial:
- Python: [Python 3.10.8 & 3.11.0]

Can it be installed successfully in these versions of Python on your PC?
Thank you very much for your attention.

**Comment by cbravo93** [comment]
Hi @ShaoZhiting24 !

At the moment we have only tested with Python 3.8, but the error you get is a bit odd. What is your pip version?

C

**Comment by ShaoZhiting24** [comment]
Hi @cbravo93 

Thank you for your attention.
My pip version is 22.3.1.

Best,
Shao

**Comment by ShaoZhiting24** [comment]
My Python version is 3.10.8.

**Comment by cbravo93** [comment]
Can you double check this:

```
ls /usr/bin/python*
```

Did you have a previous Python installation? I am suspecting you are loading that one, instead of the new ones. Can you show the output of:

```
python --version
```

**Comment by ShaoZhiting24** [comment]
Thank you, I've checked:
```
>>>ls /usr/bin/python*
/usr/bin/python2    /usr/bin/python3    /usr/bin/python3.6-config  /usr/bin/python3.6m-config         /usr/bin/python3-config
/usr/bin/python2.7  /usr/bin/python3.6  /usr/bin/python3.6m        /usr/bin/python3.6m-x86_64-config

```

And
```
>>>python --version
Python 3.10.8
```
One thing I may have to let you know, is that this python is installed in my environment in Linux, which is used to run pyscenic, therefore I installed version 3.10.8 here.

**Comment by cbravo93** [comment]
I think it is using your python3.6 version in the installation.  Can you try:

```
python3 --version
python3 -m pip install .
```

**Comment by ShaoZhiting24** [comment]
I'm so sorry that it still reported the same error...
Maybe cannot work in my present environment?

**Comment by cbravo93** [comment]
I have successfully installed it in Python 3.10.8... I think the safest will be that you start a new environment, making sure it is not using you python 3.6 installation.

**Comment by ShaoZhiting24** [comment]
Sure. Thank you very much for your help!

**Comment by cbravo93** [comment]
For reference, I created a singularity image as below:

```
FROM python:3.10-slim AS compile-image

ENV DEBIAN_FRONTEND=noninteractive
RUN BUILDPKGS="build-essential \
        libcurl4-openssl-dev \
        zlib1g-dev \
        libfftw3-dev \
        libc++-dev \
        git \
        wget \
        " && \
    apt-get update && \
    apt-get install -y --no-install-recommends apt-utils debconf locales locales-all && dpkg-reconfigure locales && \
    apt-get install -y --no-install-recommends $BUILDPKGS

RUN python -m venv /opt/venv
# Make sure we use the virtualenv:
ENV PATH="/opt/venv/bin:$PATH"

# install dependencies:
COPY pycisTopic/requirements.txt /tmp/
RUN pip install --no-cache-dir --upgrade pip wheel && \
    pip install --no-cache-dir Cython numpy && \
    pip install --no-cache-dir fitsne && \
    pip install --no-cache-dir papermill && \
    pip install --no-cache-dir igv_jupyterlab && \
    pip install --no-cache-dir bs4 && \
    pip install --no-cache-dir MACS2 && \
    pip install --no-cache-dir lxml && \
    pip install --no-cache-dir tspex && \
    pip install --no-cache-dir -r /tmp/requirements.txt
    
# install ctxcore from local copy:
COPY ctxcore /tmp/ctxcore
RUN  cd /tmp/ctxcore && \
     pip install . && \
     cd .. && rm -rf ctxcore

# install pycisTopic from local copy:
COPY pycisTopic /tmp/pycisTopic
RUN  cd /tmp/pycisTopic && \
     pip install . && \
     cd .. && rm -rf pycisTopic

# install Mallet (https://github.com/mimno/Mallet)
# https://github.com/docker-library/openjdk/blob/0584b2804ed12dca7c5e264b5fc55fc07a3ac148/8-jre/slim/Dockerfile#L51-L54
RUN mkdir -p /usr/share/man/man1 && \
    apt-get install -y --no-install-recommends ant openjdk-11-jdk && \
    git clone --depth=1 https://github.com/mimno/Mallet.git /tmp/Mallet && \
    cd /tmp/Mallet && \
    ant

# install pycistarget
COPY pycistarget /tmp/pycistarget
RUN cd /tmp/pycistarget && \
    pip install . && \
    cd .. && rm -rf pycistarget

FROM python:3.10-slim AS build-image

RUN mkdir -p /usr/share/man/man1 && \
    apt-get -y update && \
    apt-get -y --no-install-recommends install \
        openjdk-11-jdk \
        procps \
        bash-completion \
        curl \
        libfftw3-dev \
        less && \
    rm -rf /var/cache/apt/* && \
    rm -rf /var/lib/apt/lists/*

COPY --from=compile-image /opt/venv /opt/venv
COPY --from=compile-image /tmp/Mallet /opt/mallet

# Make sure we use the virtualenv:
ENV PATH="/opt/venv/bin:$PATH"
ENV PATH="/opt/mallet/bin:$PATH"

```

**Comment by cbravo93** [comment]
Oh, I eventually got the error later on, but it is related to MACS2! Sorry, I didnt see this in your output and thought it came directly from here:

```
Collecting MACS2
  Downloading MACS2-2.2.7.1.tar.gz (2.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.1/2.1 MB 33.9 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [1 lines of output]
      CRITICAL: Python version must >= 3.6!
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

```

The installation works in 3.8 with no issues, I'll open an issue in https://github.com/macs3-project/MACS.

**Comment by ShaoZhiting24** [comment]
Got it! I would use Python 3.8 for it then.
Thank you very much for your kind help!

**Comment by cbravo93** [comment]
Thanks for the heads up! I opened an issue in MACS: https://github.com/macs3-project/MACS/issues/535; Ill keep an eye and recheck that if this is solved the package installs correctly in >3.8.

Cheers!

C
